src/lib/storage/table.cpp

```diff
@@ -75,8 +76,35 @@ std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) { return _chunks.at(ch
 std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
 
 void Table::compress_chunk(const ChunkID chunk_id) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+
+  DebugAssert(chunk_id < chunk_count(), "invalid chunk id "+std::to_string(chunk_id)+". table only has "+std::to_string(chunk_count())+" chunks");
+
+  auto old_chunk = get_chunk(chunk_id);
```

Comment:
``get_chunk()`` will already perform a range check, so you don't need the assert here (except for the error message being a little nicer I guess)

Reply:
We go with the nicer error message and the debugAssert because it will not be executed in the final release build.